File: 第三方/选择图片/swift和OC图片选择/AlbumPicker-master/AlbumPicker/AlbumPicker/LSYAlbumPicker_OC/View/LSYAssetPreviewView.m
//
//  LSYAssetPreviewView.m
//  AlbumPicker
//
//  Created by Labanotation on 15/8/1.
//  Copyright (c) 2015年 okwei. All rights reserved.
//

#import "LSYAssetPreviewView.h"
#import "LSYPickerButtomView.h"
@interface LSYAssetPreviewNavBar()
@property (nonatomic,strong) UIButton *backButton;

@end
@implementation LSYAssetPreviewNavBar
- (instancetype)initWithCoder:(NSCoder *)coder
{
    self = [super initWithCoder:coder];
    if (self) {
        [self initPreviewNavBar];
    }
    return self;
}
-(instancetype)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        [self initPreviewNavBar];
    }
    return self;
}
-(void)initPreviewNavBar
{
    [self addSubview:self.selectButton];
    [self addSubview:self.backButton];
}
-(UIButton *)selectButton
{
    if (!_selectButton) {
        _selectButton = [[UIButton alloc] init];
        [_selectButton setImage:[UIImage imageNamed:@"AlbumPicker.bundle/FriendsSendsPicturesSelectBigNIcon@2x"] forState:UIControlStateNormal];
        [_selectButton setImage:[UIImage imageNamed:@"AlbumPicker.bundle/FriendsSendsPicturesSelectBigYIcon@2x"] forState:UIControlStateSelected];
        [_selectButton setSelected:YES];
        [_selectButton addTarget:self action:@selector(buttonClick:) forControlEvents:UIControlEventTouchUpInside];
    }
    return _selectButton;
}
-(UIButton *)backButton
{
    if (!_backButton) {
        _backButton = [[UIButton alloc] init];
        [_backButton setImage:[UIImage imageNamed:@"AlbumPicker.bundle/barbuttonicon_back"] forState:UIControlStateNormal];
        [_backButton addTarget:self action:@selector(buttonClick:) forControlEvents:UIControlEventTouchUpInside];
    }
    return _backButton;
}
-(void)buttonClick:(UIButton *)sender
{
    if (sender == self.selectButton) {
        if (self.delegate && [self.delegate respondsToSelector:@selector(selectButtonClick:)]) {
            [self.delegate selectButtonClick:sender];
        }
    }
    else if (sender == self.backButton){
        if (self.delegate && [self.delegate respondsToSelector:@selector(backButtonClick:)]) {
            [self.delegate backButtonClick:sender];
        }
    }
}
-(void)layoutSubviews
{
    [super layoutSubviews];
    [self.selectButton setFrame:CGRectMake(ViewSize(self).width-60, (ViewSize(self).height-40)/2, 60, 40)];
    [self.backButton setFrame:CGRectMake(0, (ViewSize(self).height-40)/2, 40, 40)];
}
/*
// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    // Drawing code
}
*/

@end


/**
 assetPreViewToolBar
 */
@interface LSYAssetPreviewToolBar ()
@property (nonatomic,strong) LSYSendButton *sendButton;
@end
@implementation LSYAssetPreviewToolBar
- (instancetype)initWithCoder:(NSCoder *)coder
{
    self = [super initWithCoder:coder];
    if (self) {
        [self initPreviewToolBar];
    }
    return self;
}
-(instancetype)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        [self initPreviewToolBar];
    }
    return self;
}
-(void)initPreviewToolBar
{
    [self addSubview:self.sendButton];
}
-(LSYSendButton *)sendButton
{
    if (!_sendButton) {
        _sendButton = [[LSYSendButton alloc] init];
        [_sendButton setTitle:@"发送" forState:UIControlStateNormal];
        [_sendButton addTarget:self action:@selector(buttonClick:) forControlEvents:UIControlEventTouchUpInside];
        
    }
    return _sendButton;
}
-(void)buttonClick:(UIButton *)sender
{
    if (sender == self.sendButton) {
        if (self.delegate && [self.delegate respondsToSelector:@selector(sendButtonClick:)]) {
            [self.delegate sendButtonClick:sender];
        }
    }
    
}
-(void)setSendNumber:(int)number
{
    [self.sendButton setSendNumber:number];
}
-(void)layoutSubviews
{
    [super layoutSubviews];
    [self.sendButton setFrame:CGRectMake(ViewSize(self).width-80, 0, 80, ViewSize(self).height)];
}
@end

File: 第三方/选择图片/swift和OC图片选择/AlbumPicker-master/AlbumPicker_Swift/AlbumPicker_Swift/LSYAlbumPicker_Swift/View/LSYAssetPreviewView.swift
//
//  LSYAssetPreviewView.swift
//  AlbumPicker
//
//  Created by okwei on 15/8/6.
//  Copyright (c) 2015年 okwei. All rights reserved.
//

import Foundation
import UIKit
import AssetsLibrary
protocol LSYAssetPreviewNavBarDelegate:class{
    func selectButtonClick(selectButton:UIButton)->Void
    func backButtonClick(backButton:UIButton)->Void
}
protocol LSYAssetPreviewToolBarDelegate:class{
    func sendButtonClick(sendButton:UIButton)->Void
}
class LSYAssetPreviewNavBar: UIView {
    var selectButton:UIButton!{
        didSet{
            selectButton.setImage(UIImage(named: "AlbumPicker.bundle/FriendsSendsPicturesSelectBigNIcon@2x"), forState: UIControlState.Normal)
            selectButton.setImage(UIImage(named: "AlbumPicker.bundle/FriendsSendsPicturesSelectBigYIcon@2x"), forState: UIControlState.Selected)
            selectButton.selected = true
            selectButton.addTarget(self, action: #selector(LSYAssetPreviewNavBar.buttonClick(_:)), forControlEvents: UIControlEvents.TouchUpInside)
            self.addSubview(selectButton)
        }
    }
    weak var delegate:LSYAssetPreviewNavBarDelegate!
    private var backButton:UIButton!{
        didSet{
            backButton.setImage(UIImage(named: "AlbumPicker.bundle/barbuttonicon_back"), forState: UIControlState.Normal)
            backButton.addTarget(self, action: #selector(LSYAssetPreviewNavBar.buttonClick(_:)), forControlEvents: UIControlEvents.TouchUpInside)
            self.addSubview(backButton)
        }
    }
    override init(frame: CGRect) {
        super.init(frame: frame)
        self.setup();
    }
    required init(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)!
        self.setup();
    }
    private func setup(){
        self.selectButton = UIButton()
        self.backButton = UIButton()

    }
    func buttonClick(sender:UIButton){
        if sender == self.selectButton{
            if self.delegate != nil {
                self.delegate.selectButtonClick(sender)
            }
        }
        else if sender == self.backButton{
            self.delegate.backButtonClick(sender)
        }
    }
    override func layoutSubviews() {
        super.layoutSubviews()
        self.selectButton.frame = CGRectMake(LSYSwiftDefine.ViewSize(self).width-60, (LSYSwiftDefine.ViewSize(self).height-40)/2, 60, 40)
        self.backButton.frame = CGRectMake(0, (LSYSwiftDefine.ViewSize(self).height-40)/2, 40, 40)
    }
}
class LSYAssetPreviewToolBar: UIView {
    weak var delegate:LSYAssetPreviewToolBarDelegate!
    private var sendButton:LSYSendButton!{
        didSet{
            sendButton.setTitle("发送", forState: UIControlState.Normal)
            sendButton.addTarget(self, action: #selector(LSYAssetPreviewNavBar.buttonClick(_:)), forControlEvents: UIControlEvents.TouchUpInside)
            self.addSubview(sendButton)
        }
    }
    func buttonClick(sender:UIButton){
        if sender == self.sendButton{
            if self.delegate != nil{
                self.delegate.sendButtonClick(sender)
            }
        }
    }
    func setSendNumber(number:Int){
        self.sendButton.setSendNumber(number)
    }
    override init(frame: CGRect) {
        super.init(frame: frame)
        self.setup()
    }
    required init(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)!
        self.setup()
    }
    func setup(){
        self.sendButton = LSYSendButton()
    }
    override func layoutSubviews() {
        super.layoutSubviews()
        self.sendButton.frame = CGRectMake(LSYSwiftDefine.ViewSize(self).width-80, 0, 80, LSYSwiftDefine.ViewSize(self).height)
    }
}
//MARK:-
//MARK:LSYAssetPreviewItem
protocol LSYAssetPreviewItemDelegate:class{
    func hiddenBarControl()->Void
}

class LSYAssetPreviewItem: UIView {
    
    weak var delegate:LSYAssetPreviewItemDelegate!
    var asset:ALAsset!{
        didSet{
            assetImageView.image = UIImage(CGImage: asset.defaultRepresentation().fullResolutionImage().takeUnretainedValue())
            let imageViewHeight:CGFloat = (assetImageView.image?.size.height)! / (assetImageView.image?.size.width)! * previewScrollView.frame.size.width
            let imageViewWidth = previewScrollView.frame.size.width
            assetImageView.frame = CGRectMake(0, 0, imageViewWidth, imageViewHeight)
            assetImageView.center = CGPointMake(assetImageView.center.x, CGRectGetMidY(self.previewScrollView.frame))
        }
    }
    var previewScrollView:UIScrollView!{
        didSet{
            previewScrollView.center = CGPointMake(previewScrollView.center.x, CGRectGetMidY(self.frame))
            previewScrollView.delegate = self
            previewScrollView.showsHorizontalScrollIndicator = false
            previewScrollView.showsVerticalScrollIndicator = false
            previewScrollView.minimumZoomScale = 1.0
            previewScrollView.maximumZoomScale = 2.0
            self.addSubview(previewScrollView)
        }
    }
   
     var assetImageView:UIImageView!{
        didSet{
            self.previewScrollView.addSubview(assetImageView)
        }
    }
    override init(frame: CGRect) {
        super.init(frame: frame)
        self.setup()
    }
    required init(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)!
        self.setup()
    }
    private func setup(){
        self.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(LSYAssetPreviewItem.hiddenBar)))
        self.previewScrollView = UIScrollView(frame: CGRectMake(10, 0, LSYSwiftDefine.ViewSize(self).width-20, LSYSwiftDefine.ViewSize(self).height))
        self.assetImageView = UIImageView()
        
    }
    func hiddenBar(){
        if self.delegate != nil {
            self.delegate.hiddenBarControl()
        }
    }
}